### ISSUE TITLE
4 out of my 10 self hosted runners appear offline, but start up correctly.

### ISSUE THREAD

**Issue by nixomose** [actions/runner#3368]
**Describe the bug**
All my linux and windows self hosted runners have been running fine for months and all of a sudden 4 of them say "Offline" and won't take work, but they're up and listening with no errors in the logs

**To Reproduce**
Start self hosted github runner with 

C:\actions-runner>run.cmd
        1 file(s) copied.

√ Connected to GitHub

Current runner version: '2.311.0'
2024-06-30 22:35:46Z: Listening for Jobs

**Expected behavior**
github runner status page should show the runner as Idle but instead it says "Offline"

## Runner Version and Platform
Version of your runner?
Current runner version: '2.311.0'

OS of the machine running the runner? OSX/Windows/Linux/...
I have windows and linux runners doing the same thing.

## What's not working?
Please include error messages and screenshots.
There are no errors, that's the problem.
I ran config --check and everything passes.
I am afraid to restart my running workers in case they don't come back. 6 of my 10 are fine, 4 are exhibiting this behavior.

## Job Log Output
There is no job, because the runner doesn't accept requests.
Some of my logs say this:

[2024-06-30 21:19:23Z INFO RSAEncryptedFileKeyManager] Loading RSA key parameters from file C:\actions-runner\.credentials_rsaparams
[2024-06-30 21:19:23Z INFO RSAEncryptedFileKeyManager] Loading RSA key parameters from file C:\actions-runner\.credentials_rsaparams
[2024-06-30 21:19:23Z INFO GitHubActionsService] AAD Correlation ID for this token request: Unknown
[2024-06-30 21:19:23Z INFO MessageListener] Session created.
[2024-06-30 21:19:23Z INFO Terminal] WRITE LINE: Current runner version: '2.311.0'
[2024-06-30 21:19:23Z INFO Terminal] WRITE LINE: 2024-06-30 21:19:23Z: Listening for Jobs
[2024-06-30 21:19:23Z INFO JobDispatcher] Set runner/worker IPC timeout to 30 seconds.
[2024-06-30 21:49:25Z INFO MessageListener] No message retrieved from session 'b4a34cfd-9a47-4c57-bf2a-d1e6f4b80d4a' within last 30 minutes.
[2024-06-30 22:14:26Z INFO RSAEncryptedFileKeyManager] Loading RSA key parameters from file C:\actions-runner\.credentials_rsaparams
[2024-06-30 22:14:26Z INFO GitHubActionsService] AAD Correlation ID for this token request: Unknown
[2024-06-30 22:19:27Z INFO MessageListener] No message retrieved from session 'b4a34cfd-9a47-4c57-bf2a-d1e6f4b80d4a' within last 30 minutes.
[2024-06-30 22:33:27Z INFO HostContext] Runner will be shutdown for UserCancelled
[2024-06-30 22:33:27Z WARN GitHubActionsService] GET request to https://pipelinesghubeus7.actions.githubusercontent.com/mCNgUk0kyXGZzsNSxNubKPNpa6onWSrSkvn1B09gas4P4iFzu6/_apis/distributedtask/pools/1/messages?sessionId=b4a34cfd-9a47-4c57-bf2a-d1e6f4b80d4a&status=Online&runnerVersion=2.311.0&os=Windows&architecture=X64 has been cancelled.
[2024-06-30 22:33:27Z INFO MessageListener] Get next message has been cancelled.
[2024-06-30 22:33:27Z INFO JobDispatcher] Shutting down JobDispatcher. Make sure all WorkerDispatcher has finished.
[2024-06-30 22:33:27Z INFO Listener] Runner execution been cancelled.

## Runner and Worker's Diagnostic Logs
If applicable, add relevant diagnostic log information.  Logs are located in the runner's `_diag` folder. The runner logs are prefixed with `Runner_` and the worker logs are prefixed with `Worker_`. Each job run correlates to a worker log.  All sensitive information should already be masked out, but please double-check before pasting here.

**Comment by luketomlinson** [comment]
HI @nixomose, I noticed the runner in question is a pretty old version `2.311.0`. Can you try updating to the latest and see if that fixes your issue?

**Comment by nixomose** [comment]
yeah, sorry that's what I was doing today. the runner didn't update itself, when I manually updated to 2.317 it started working so that's that.
For windows anyway, I'll go see why the linux ones have problems, but at least I know I can fix it by upgrading, so I'm good. I'll close the ticket. Thanks for getting back to me.

At some point I'll check and see why the update failed, but I think I saw some errors to that effect yesterday. Anyway no worries, it's working and I know how to fix the ones that are broken, that's all I care about at the moment.
Thanks again.
